odoo/upgrade_code/18.1-01-rename-class.py

```diff
@@ -0,0 +1,181 @@
+import re
+
+
+def upgrade(file_manager):
+    nb = len([1 for file in file_manager if file.path.name.endswith('.py')])
+    i = 0
+    file_manager.print_progress(0, nb)
+
+    for file in file_manager:
+        if not file.path.name.endswith('.py'):
+            continue
+
+        class_name_models = get_class_name_models(file)
+        content = file.content
+        content = update_class_name(content, class_name_models)
+        content = remove_name_and_update_inherit(content)
+        content = class_with_2_blank_lines(content)
+        file.content = content
+
+        i += 1
```

Comment:
Wouldn't it be cleaner to do something like
```python
python_files = [file for file in file_manager if file.path.name.endswith('.py')]
nb = len(python_files)
for i, file in enumerate(python_files):
    ...
```

In order to
* avoid filtering twice the same thing
* manage the increment of `i` manually

Unless you are worried about memory? (but it's only a subset of the memory used by `FileManager`) :shrug: